core/persist/PersistenceProvider.ts

```diff
@@ -32,100 +39,162 @@ import {throwIf} from '@xh/hoist/utils/js';
-                return new CustomProvider(rest);
-            default:
-                throw XH.exception(`Unknown Persistence Provider for type: ${type}`);
-        }
-    }
+        let {type, ...rest} = persistOptions,
+            ret: PersistenceProvider<S>;
 
-    /**
-     * Called by implementations only. See create.
-     */
-    protected constructor({path, debounce = 250}: PersistOptions) {
-        throwIf(isUndefined(path), 'Path not specified in PersistenceProvider.');
-        this.path = path;
-        this.debounce = debounce;
-        if (debounce) {
-            this.writeInternal = isNumber(debounce)
-                ? lodashDebounce(this.writeInternal, debounce)
-                : lodashDebounce(this.writeInternal, debounce.interval, debounce);
+        try {
+            if (!type) {
+                if (rest.prefKey) type = 'pref';
+                if (rest.localStorageKey) type = 'localStorage';
+                if (rest.dashViewModel) type = 'dashView';
+                if (rest.getData || rest.setData) type = 'custom';
+            }
+
+            switch (type) {
+                case 'pref':
+                    ret = new PrefProvider(cfg);
+                    break;
+                case 'localStorage':
+                    ret = new LocalStorageProvider(cfg);
+                    break;
+                case `dashView`:
+                    ret = new DashViewProvider(cfg);
+                    break;
+                case 'custom':
+                    ret = new CustomProvider(cfg);
+                    break;
+                default:
+                    throw XH.exception(`Unknown Persistence Provider for type: ${type}`);
+            }
+
+            ret.bindToTarget(target);
+            return ret;
+        } catch (e) {
+            logError(e, cfg.owner ?? 'PersistenceProvider');
+            ret?.destroy();
+            return null;
         }
     }
 
     /**
-     * Read data at a path
+     * Read persisted data at a path
      */
-    read(): any {
-        return get(this.readRaw(), this.path);
+    read(): PersistableState<S> {
+        const state = get(this.readRaw(), this.path);
+        return !isUndefined(state) ? new PersistableState(state) : null;
     }
 
     /**
-     * Save data at a path
-     * @param data  - data to be written to the path, must be serializable to JSON.
+     * Persist data to a path.
+     * @param data  - data to be written to the path, must be JSON serializable.
      */
-    write(data: any) {
+    write(data: S) {
         this.writeInternal(data);
     }
 
-    /**
-     * Clear any state saved by this object at a path
-     */
+    /** Clear any persisted data at a path. */
     clear(path: string = this.path) {
         const obj = cloneDeep(this.readRaw());
         unset(obj, this.path);
         this.writeRaw(obj);
     }
 
-    /**
-     * Clear *all* state held by this object.
-     */
+    /** Clear *all* persisted data managed by this provider. */
```

Comment:
This API doesn't sit well with me - unclear why we have it, and I don't think the doc comment is correct.  

This method doesn't just clear the data this particular provider manages - that's `clear()` above, which is scoped to the provider's own path. Instead `clearAll()` clears.... all data held by the same backing implementation, across all paths, I think....  Is that something that we want/need to be able to call on a single provider?